sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/BinaryDataHelper.java

```diff
@@ -50,8 +52,14 @@ public static void setAccessor(final BinaryDataAccessor binaryDataAccessor) {
      * @throws NullPointerException If {@code content} is null.
      */
     public static BinaryData createBinaryData(BinaryDataContent content) {
-        ensureAccessorSet();
-        return accessor.createBinaryData(content);
+        try {
+            ensureAccessorSet();
+            return accessor.createBinaryData(content);
+        } catch (Exception e) {
+            LOGGER.error("createBinary failed with message : {}", e.getMessage());
```

Comment:
You don't need the second logging line - just pass e as second parameter and Log4J will log the stack trace.